joblib/test/test_memory.py

```diff
@@ -162,14 +162,12 @@ def name_collision(x):
 
     b = name_collision
 
-    with warnings.catch_warnings(record=True) as w:
-        # Cause all warnings to always be triggered.
-        warnings.simplefilter("always")
+    with warns(JobLibCollisionWarning) as warninfo:
```

Comment:
The pytest use `record`, I would be in favour of using the same naming everywhere unless you have seen `warninfo` used a lot in your experience with pytest.